### DISCUSSION TITLE
Role Function

### DISCUSSION THREAD

**Discussion by BritneyStars** [apache/skywalking#10262] [Q&A] [answered]
I find some questions about  Aggregator role . In document ,Aggregator mode OAP receives data from {@link #Mixed} and {@link #Aggregator} OAP nodes, and do 2nd round  aggregation. Then save the final result to the storage. but in fact ,I found Aggregator  mode also do 1nd round  aggregation. In code,
if (CoreModuleConfig.Role.Mixed.name()
                                       .equalsIgnoreCase(
                                           moduleConfig.getRole())
            || CoreModuleConfig.Role.Aggregator.name()
                                               .equalsIgnoreCase(
                                                   moduleConfig.getRole())) {
            RemoteInstance gRPCServerInstance = new RemoteInstance(gRPCServerInstanceAddress);
            coordinator.registerRemote(gRPCServerInstance);
        }
whether shows that the mixed role and Aggregator role' function are same, all do L1 and L2 aggregator

**Comment by wu-sheng** [comment, marked as answer]
Whether do or don't do L1 aggregation relies on whether you open receiver or fetcher.
As the modulization design principal, we don't have point to limit to do so.
The role determines the aggregation data flow, and how to reflush data into Database.

**Reply by BritneyStars** [reply]
Thanks

**Reply by Norman-else** [reply]
@wu-sheng Hi, Mr Wu. I checked the source code and found that there are many modules that rely on the sharing-receiver module, so my question is: for the aggregator, do I need to disable all modules related to the sharing receiver to turn off the receiving of the aggregator?
